static/src/javascripts/lib/cmp.spec.js

```diff
@@ -0,0 +1,215 @@
+        expect(myCallBack).toHaveBeenCalledTimes(1);
+    });
+
+    describe('consentState', () => {
+        it('returns null state if cmp lib has not been initialised', () => {
+            expect(consentState('functional')).toBeNull();
+            expect(consentState('performance')).toBeNull();
+            expect(consentState('advertisement')).toBeNull();
+        });
+
+        it('returns functional consent state', () => {
+            init();
+
+            expect(consentState('functional')).toBe(true);
+        });
+
+        it('returns performance consent state', () => {
+            init();
+
+            expect(consentState('performance')).toBe(true);
+        });
+
+        it('returns advertisement consent state with true if getAdConsentState true', () => {
+            getAdConsentState.mockReturnValue(true);
+
+            init();
+
+            expect(consentState('advertisement')).toBe(true);
+        });
+
+        it('returns advertisement consent state with false if getAdConsentState true', () => {
+            getAdConsentState.mockReturnValue(false);
+
+            init();
+
+            expect(consentState('advertisement')).toBe(false);
+        });
+    });
+
+    describe('onConsentNotification', () => {
+        describe('if cmpIsReady is TRUE when onConsentNotification called', () => {
+            it('executes functional callback', () => {
+                const myCallBack = jest.fn();
+
+                init();
+
+                onConsentNotification('functional', myCallBack);
+
+                expect(myCallBack).toHaveBeenCalledTimes(1);
+                expect(myCallBack).toBeCalledWith(true);
+            });
+
+            it('executes performance callback', () => {
+                const myCallBack = jest.fn();
+
+                init();
+
+                onConsentNotification('performance', myCallBack);
+
+                expect(myCallBack).toHaveBeenCalledTimes(1);
+                expect(myCallBack).toBeCalledWith(true);
+            });
+
+            it('executes advertisement callback with true if getAdConsentState true', () => {
+                getAdConsentState.mockReturnValue(true);
+
+                const myCallBack = jest.fn();
+
+                init();
+
+                onConsentNotification('advertisement', myCallBack);
+
+                expect(myCallBack).toHaveBeenCalledTimes(1);
+                expect(myCallBack).toBeCalledWith(true);
+            });
+
+            it('executes advertisement callback with true if getAdConsentState false', () => {
+                getAdConsentState.mockReturnValue(false);
+
+                const myCallBack = jest.fn();
+
+                init();
+
+                onConsentNotification('advertisement', myCallBack);
+
+                expect(myCallBack).toHaveBeenCalledTimes(1);
+                expect(myCallBack).toBeCalledWith(false);
+            });
+
+            it('executes advertisement callback each time consent nofication triggered', () => {
```

Comment:
This would be a great candidate for property based testing, where given a random number, `n`, we check that `n` re-triggers leads to `n + 1` invocations. At the moment we're only checking it can be re-triggered once. @janua found a good library for this sort of thing in Typescript and may have tips here!

Reply:
Interesting idea, I'll have a go at implementing something that does this!

Reply:
That's a great idea!